RecoTauTag/HLTProducers/interface/L1TJetsMatching.h

```diff
@@ -0,0 +1,207 @@
+            
+            if ((mjj > Mjj) && (myJet1.pt() >= pt1) && (myJet2.pt() > pt2) )
+            {
+                
+                output.first.push_back(myJet1);
+                output.first.push_back(myJet2);
+                
+            }
+            
+            if ((mjj > Mjj) && (myJet1.pt() < pt1) && (myJet1.pt() > pt2) && (myJet2.pt() > pt2))
+            {
+                
+                const T &  myJetTest = (pfMatchedJets)[0];
+                if (myJetTest.pt()>pt1){
+                    output.second.push_back(myJet1);
+                    output.second.push_back(myJet2);
+                    output.second.push_back(myJetTest);
+                    
+                }
+            }
+            
+        }
+        
+        return output;
+        
+    }
+    template< typename T>
+    L1TJetsMatching<T>::L1TJetsMatching(const edm::ParameterSet& iConfig):
+    jetSrc_    ( consumes<std::vector<T>>                     (iConfig.getParameter<InputTag>("JetSrc"      ) ) ),
+    jetTrigger_( consumes<trigger::TriggerFilterObjectWithRefs>(iConfig.getParameter<InputTag>("L1JetTrigger") ) ),
+    pt1_Min   ( iConfig.getParameter<double>("pt1_Min")),
+    pt2_Min   ( iConfig.getParameter<double>("pt2_Min")),
+    mjj_Min   ( iConfig.getParameter<double>("mjj_Min"))
+    {
+        produces<std::vector<T>>("TwoJets");
+        produces<std::vector<T>>("ThreeJets");
+    }
+    template< typename T>
+    L1TJetsMatching<T>::~L1TJetsMatching(){ }
+    
+    template< typename T>
+    void L1TJetsMatching<T>::produce(edm::StreamID iSId, edm::Event& iEvent, const edm::EventSetup& iES) const
+    {
+        
+        unique_ptr<std::vector<T>> pfMatchedJets(new std::vector<T>);
+        std::pair<std::vector<T>,std::vector<T>> output;
+        
+        double deltaR    = 1.0;
+        double matchingR = 0.5;
+        
+        // Getting HLT jets to be matched
+        edm::Handle<std::vector<T> > pfJets;
+        iEvent.getByToken( jetSrc_, pfJets );
+        
+        edm::Handle<trigger::TriggerFilterObjectWithRefs> l1TriggeredJets;
+        iEvent.getByToken(jetTrigger_,l1TriggeredJets);
+        
+        //l1t::TauVectorRef jetCandRefVec;
+        l1t::JetVectorRef jetCandRefVec;
+        l1TriggeredJets->getObjects( trigger::TriggerL1Jet,jetCandRefVec);
+        
+        math::XYZPoint a(0.,0.,0.);
+        
+        //std::cout<<"PFsize= "<<pfJets->size()<<endl<<" L1size= "<<jetCandRefVec.size()<<std::endl;
+        for(unsigned int iJet = 0; iJet < pfJets->size(); iJet++){
+            const T &  myJet = (*pfJets)[iJet];
+            for(unsigned int iL1Jet = 0; iL1Jet < jetCandRefVec.size(); iL1Jet++){
+                // Find the relative L2pfJets, to see if it has been reconstructed
+                //  if ((iJet<3) && (iL1Jet==0))  std::cout<<myJet.p4().Pt()<<" ";
+                deltaR = reco::deltaR2(myJet.p4().Vect(), (jetCandRefVec[iL1Jet]->p4()).Vect());
```

Comment:
I think deltaR2 returns deltaR^2... please check and the square the cut value (one at the beginning) and compare, rather than taking the sqrt, to be faster.